pkg/portal/cluster/machines.go

```diff
@@ -59,7 +65,52 @@ func (c *client) Machines(ctx context.Context) (*MachineListInformation, error)
 	return c.fetcher.Machines(ctx)
 }
 
+func (c *client) VMAllocationStatus(ctx context.Context) (map[string]string, error) {
+	return c.fetcher.VMAllocationStatus(ctx)
+}
+
+func (f *realFetcher) VMAllocationStatus(ctx context.Context) (map[string]string, error) {
+	env := f.azureSideFetcher.env
+	subscriptionDoc := f.azureSideFetcher.subscriptionDoc
+	clusterRGName := f.azureSideFetcher.resourceGroupName
+	fpAuth, err := env.FPAuthorizer(subscriptionDoc.Subscription.Properties.TenantID, env.Environment().ResourceManagerEndpoint)
+	if err != nil {
+		return nil, err
+	}
+	// Getting Virtual Machine resources through the Cluster's Resource Group
+	computeResources, err := f.resourceFactory.NewResourcesClient(env.Environment(), subscriptionDoc.ID, fpAuth).ListByResourceGroup(ctx, clusterRGName, "resourceType eq 'Microsoft.Compute/virtualMachines'", "", nil)
+	if err != nil {
+		return nil, err
+	}
+	vmAllocationStatus := make(map[string]string)
+	virtualMachineClient := f.resourceFactory.NewVirtualMachinesClient(env.Environment(), subscriptionDoc.ID, fpAuth)
+	for _, res := range computeResources {
+		putAllocationStatusToMap(ctx, clusterRGName, vmAllocationStatus, res, virtualMachineClient, f.log)
+	}
+
+	return vmAllocationStatus, nil
+}
+
 // Helper Functions
+func putAllocationStatusToMap(ctx context.Context, clusterRGName string, vmAllocationStatus map[string]string, res mgmtfeatures.GenericResourceExpanded, virtualMachineClient compute.VirtualMachinesClient, log *logrus.Entry) {
+	var vmName, allocationStatus string
+	vm, err := virtualMachineClient.Get(ctx, clusterRGName, *res.Name, mgmtcompute.InstanceView)
+	if err != nil {
+		log.Warn(err) // can happen when the ARM cache is lagging
+		return
+	}
+
+	vmName = *vm.Name
+	instanceViewStatuses := vm.InstanceView.Statuses
+	for _, status := range *instanceViewStatuses {
+		if strings.HasPrefix(*status.Code, "PowerState/") {
+			allocationStatus = *status.Code
+		}
+	}
+
+	vmAllocationStatus[vmName] = allocationStatus
```

Comment:
then remove that line from my suggestion and keep the rest of it

Reply:
ohh yaa, sorry, if we do not get any status from then it is going to be empty string, I did this to keep the UI clean as this is going to get seen in the UI as is.

Reply:
I just want to have all the VMNames as keys in the map.

Reply:
then change `"unknown"` to `""`

Reply:
No, all the values of the map will have empty string, I have declared the variable `allocationStatus` at the starting of the function, if there is value for the vm allocationStatus code, the var will have that value else it would be empty string only. So I think it is good as is.

Reply:
take a look again at my suggestion, it does exactly that. the `return` statement break the flow

Reply:
Yes yes .. you are right .. I am changing it to empty String.